packages/circuit-ui/components/NotificationToast/NotificationToast.docs.mdx

```diff
@@ -0,0 +1,121 @@
+
+export default function App() {
+  return <ToastProvider>{/* Your content goes here... */}</ToastProvider>;
+}
+```
+
+Then, use the `useNotificationToast` hook to open a toast from a component:
+
+```tsx
+import { useNotificationToast, Button } from '@sumup/circuit-ui';
+
+export function App({}) {
+  const { setToast } = useNotificationToast();
+
+  const handleClick = () => {
+    setToast({
+      body: 'This is a toast message',
+    });
+  };
+
+  return <Button onClick={handleClick}>Open toast</Button>;
+}
+```
+
+## Accessibility
+
+By their nature as small quick messages, or interruptions, toasts map to [live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions).
+Live region must always exist on the page at render, before the toast is generated.
+
+Ideally, the toast messages should not contain any interactive elements.
+The screen readers are gonna announce the raw text within toast messages, without structure and any need to move the user's focus.
+Keyboard users won’t be able to move to the interactive elements with a clear path.
+For example, there is a toast message with the text “You successfully updated your data.”, followed by a link or button with the text "Back".
+Screen readers are gonna communicate a message as “You successfully updated your data. Back.”, and the semantics of the link "Back" is not announced.
+
+To match the content, the role and aria-live level is adapted. The role is set to `"status"`, and it has `aria-live="polite"` meaning the screen reader will wait before announcing the update.
+
+### Best practices
+
+This component adheres to the following principles from the Web Content Accessibility Guidelines (WCAG):
+
+#### Toast duration
+
+The auto-dimiss duration of the toast is adjustable.
+Based on how fast the average American reads, a good lenght of time to keep the toast message up is 5 seconds plus 1 extra second for every 120 words.
+The shortest default that should be used as a best practice is 6 seconds.
+
+#### Consistent identification
+
+Based on the WCAG guideline [3.2.4](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html) each component should be consistent with another.
+
+#### Location
+
+Following consistent identificaton, toast messages should be also placed consistently. Generally toast messages usually appear at the start or end of the DOM. If you place them somewhere else, it can cause difficulties for screen reader users to locate them.
+
+#### Keyboard access and focus order
+
+For any interactive element, user should be able to tab and shift through them by using the keyboard, which means they require a keyboard focus indicator.
+Follow more in the WCAG guideline [2.1.1](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)
+
+Using the keyboard, if the toast message has any interactive elements, then it should always show up in the same place in the focus order, in a way that makes sense to a user.
```

Comment:
But the toast should never have interactive elements 🤔

Reply:
Most of the articles cover up the cases when there is interactive element for the keyboard navigation. I agree that it does not go well with the idea of the toasts not having any interactive element. I wouldn't mind deleting this part in the end if that is ok for you ?

Reply:
Yes I think that makes sense. A toast can be badly accessible if it has interactivity, but it can not be accessible if it has interactivity+auto-dismissal. In our case it has auto-dismissal, therefore it should never have interactivity 🙂 (the close button is just progressive enhancement and doesn't count as interactivity)